opensrp-anc/src/main/assets/rule/tests_calculation_rules.yml

```diff
@@ -4,284 +4,284 @@ description: hiv_positive
 actions:
-    - "calculation = 1"
+  - "calculation = 1"
 ---
 name: step2_gdm
 description: gdm
 priority: 1
 condition: "(step2_fasting_plasma_gluc >= 92 && step2_fasting_plasma_gluc < 126 ) || (step2_ogtt_fasting >= 92 && step2_ogtt_fasting < 126) || (step2_ogtt_1 >= 180 && step2_ogtt_1 < 200 ) || (step2_ogtt_2 >= 153 && step2_ogtt_2 < 200)"
 actions:
-    - "calculation = 1"
+  - "calculation = 1"
 ---
 name: step2_dm_in_preg
 description: dm_in_preg
 priority: 1
 condition: "(step2_fasting_plasma_gluc >= 126) || (step2_ogtt_fasting >= 126) || (step2_ogtt_1 >= 200) || (step2_ogtt_2 >= 200) || (step2_random_plasma >= 200)"
 actions:
-    - "calculation = 1"
+  - "calculation = 1"
 ---
 name: step1_hb_result
 description: hb_result
 priority: 1
 condition: "true"
 actions:
-    - "calculation = step1_hb_test_type == 'complete_blood_count' ? step1_cbc : step1_hb_test_type == 'hb_test_haemoglobinometer' ? step1_hb_gmeter :  step1_hb_test_type == 'hb_test_colour_scale' ? step1_hb_colour : '' "
+  - "calculation = (step1_hb_test_type == 'complete_blood_count' && step1_hb_test_type != '' && step1_cbc != '') ? step1_cbc : (step1_hb_test_type == 'hb_test_haemoglobinometer' && step1_hb_test_type != '' && step1_hb_gmeter != '') ? step1_hb_gmeter :  (step1_hb_test_type == 'hb_test_colour_scale' && step1_hb_test_type != '' && step1_hb_colour != '') ? step1_hb_colour : '' "
 ---
 name: step2_hb_result
 description: hb_result
 priority: 1
 condition: "true"
 actions:
-    - "calculation = step2_hb_test_type == 'complete_blood_count' ? step2_cbc : step2_hb_test_type == 'hb_test_haemoglobinometer' ? step2_hb_gmeter : step2_hb_test_type == 'hb_test_colour_scale' ? step2_hb_colour : '' "
+  - "calculation = (step2_hb_test_type == 'complete_blood_count' && step2_hb_test_type != '' && step2_cbc != '') ? step2_cbc : (step2_hb_test_type == 'hb_test_haemoglobinometer' && step2_hb_test_type != '' && step2_hb_gmeter != '') ? step2_hb_gmeter :  (step2_hb_test_type == 'hb_test_colour_scale' && step2_hb_test_type != '' &&  step2_hb_colour != '') ? step2_hb_colour : '' "
 ---
 name: step1_anaemic
 description: anaemic
 priority: 1
-condition: "(step1_hb_result == '' && global_pallor == 'yes') || (step1_hb_result < 11 && global_gest_age_openmrs != '' && (global_gest_age_openmrs <= 12 || global_gest_age_openmrs >= 28)) || (step1_hb_result < 10.5 && global_gest_age_openmrs != '' && (global_gest_age_openmrs > 12 && global_gest_age_openmrs < 28))"
+condition: "(step1_hb_result > 0 && step1_hb_result != '' && global_pallor == 'yes') || ((step1_hb_result > 0 && step1_hb_result < 11 && step1_hb_result != '') && global_gest_age_openmrs != '' && (global_gest_age_openmrs <= 12 || global_gest_age_openmrs >= 28)) || ((step1_hb_result > 0 && step1_hb_result < 10.5 && step1_hb_result != '') && global_gest_age_openmrs != '' && (global_gest_age_openmrs > 12 && global_gest_age_openmrs < 28))"
```

Comment:
Should you first check if the result is empty before checking for the actual int result?

Reply:
Yah let me update this. You might need to reapprove

Reply:
Updated